esp-hal/src/analog/adc/xtensa.rs

```diff
@@ -234,8 +234,8 @@ impl RegisterAccess for crate::peripherals::ADC1 {
     fn set_init_code(data: u16) {
         let [msb, lsb] = data.to_be_bytes();
 
-        crate::regi2c_write_mask!(I2C_SAR_ADC, ADC_SAR1_INITIAL_CODE_HIGH_ADDR, msb as u32);
-        crate::regi2c_write_mask!(I2C_SAR_ADC, ADC_SAR1_INITIAL_CODE_LOW_ADDR, lsb as u32);
+        crate::rom::regi2c_write_mask!(I2C_SAR_ADC, ADC_SAR1_INITIAL_CODE_HIGH_ADDR, msb as u32);
```

Comment:
This macro should probably live in an `internal_analog` or similar module, rom seems somewhat random to me - it's expanding into a rom function, sure, but if that would be the driving force behind code organization we could just have a big `rust.rs` file.

Reply:
I agree with this - should it get fixed in this PR or not?

Reply:
> This macro should probably live in an internal_analog or similar module

The only issue that `regi2c_write_mask!` is used for more than just ADC, it's also used for sleep and some other subsystems iirc.

I agree that `rom` isn't ideal, but lets not block this PR on it - it'll find it's rightful place eventually.